crypto/kem/README.md

```diff
@@ -0,0 +1,227 @@
+                                                                                  int ret = FAIL;
+
+                                                                                  EVP_PKEY_CTX *encaps_ctx = NULL;
+                                                                                  EVP_PKEY *key = NULL;
+
+                                                                                  uint8_t *ct = NULL, *ss = NULL; // ciphertext and shared secret,
+                                                                                  size_t ct_len, ss_len;          // and their lengths.
+
+
+                                                                                  // ENCAPSULATION
+                                                                                  // Set the received raw public key.
+                                                                                  key = EVP_PKEY_kem_new_raw_public_key(NID_KYBER512,
+                                                                                                                        pub_key, pub_key_len);
+                                                                                  if (!key) {
+                                                                                    goto end;
+                                                                                  }
+
+                                                                                  // Create the context.
+                                                                                  encaps_ctx = EVP_PKEY_CTX_new(key, NULL);
+                                                                                  if (!encaps_ctx) {
+                                                                                    goto end;
+                                                                                  }
+
+                                                                                  // Get the required buffer lengths and allocate the buffers.
+                                                                                  if (!EVP_PKEY_encapsulate(encaps_ctx, NULL, &ct_len, NULL, &ss_len)) {
+                                                                                    goto end;
+                                                                                  }
+                                                                                  ct = (uint8_t*) OPENSSL_malloc(ct_len);
+                                                                                  ss = (uint8_t*) OPENSSL_malloc(ss_len);
+                                                                                  if (!ct || !ss) {
+                                                                                    goto end;
+                                                                                  }
+
```

Comment:
Maybe add a comment here about generating the `ss`, or a call to RAND_bytes if you want.

Reply:
the shared secret (ss) is generated inside `EVP_PKEY_encapsulate`, no need for users to do anything.

Reply:
I don't think this was mentioned, though, in the readme text. I think it's worth mentioning. I see it mentioned at the end of the example.

Reply:
done, added to lines 25 and 29.